### ISSUE TITLE
Bug: Tor connection can't connect on TailsOS

### ISSUE THREAD

**Issue by ghost** [firoorg/campfire#2]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Wallet can't connect via its Tor settings (GUI) or by being launched with torsocks (CLI) wrapper on TailsOS.

### Expected Behavior

Tor settings in wallet working properly with TailsOS.
Being able to connect to and sync on a operating system that only allows Tor connections only (firewall) via natively in the GUI setting or by being launched with `torsocks -i ./campfire-v2.0.1.AppImage` via CLI.

### Reproduce Steps

So I first downloaded the public key (reubens.asc) and gpg verified the SHA256SUMS with good sig and that the campfire-v2.0.1.AppImage file's hash was the same. So my downloaded wallet was not corrupted or altered in transport.  

Made sure the AppImage was executable and clicked it to launch it and it loaded.
However the Tor settings were never able to connect Tor, just stuck trying to connect. \
So I then tried to debug it by launching from terminal.

`./campfire-v2.0.1.AppImage`

```
flutter: Log: [Info][2024-08-31 03:48:21.029Z]: TorSyncStatusChangedEvent fired with arg newStatus = TorConnectionStatus.connecting (TorService.start call in progress)
flutter: BUILD: DraggableSwitchButtonState
```
Nothing else after this

Next I tried `torsocks --isolate ./campfire-v2.0.1.AppImage`, which shows no connections in onion-circuits. When looking at a test wallet I generated it never syncs (cause it can't connect). It is just stuck trying to sync and goes nowhere as well.

### Environment

- Operating system and version: Tails 6.6
- Device platform and version: Linux/Debian 12
- Real device or emulator/simulator: n/a

### Further Information

I'm not quit sure if this is a Wayland issue or not? However Electrum-FIRO works for me on Tails with no issues connecting. I don't really understand Dart programming language and looked at [tor_service.dart](https://github.com/firoorg/campfire/blob/main/lib/services/tor_service.dart) and [tor_subscription.dart](https://raw.githubusercontent.com/firoorg/campfire/main/lib/widgets/tor_subscription.dart) and am not quite sure?

If this is broken on Tails then I would assume this is also broken on *stack wallet* since it appears they have the same Tor setting.

This would be great to fix this so that Campfire works on Tails. Using the Firo's Campfire wallet on Tails OS presents several compelling advantages, particularly for users who prioritize privacy, security, and anonymity in their cryptocurrency transactions. I mean glaring obvious is that both are focused on privacy. Let's also not forget that Monero GUI wallet and Feather wallet both were made for easy integration for use on specifically Tails and linux. Those are both Monero related but my introduction to using Monero and crypto was on Tails. I know several people that use Tails to store their crypto wallets or use hardware wallets with the OS.

In summary, using the Firo wallet on Tails OS offers a robust solution for individuals seeking to prioritize privacy, security, and control over their cryptocurrency transactions. The combination of Firo's advanced privacy features and Tails OS's secure, anonymous environment creates a powerful tool for anyone looking to navigate the digital financial landscape safely and discreetly. Additionally, this integration could increase awareness and usage of Firo, attracting new users who are drawn to the enhanced privacy and security that both platforms provide. By promoting a secure and private way to manage cryptocurrency, the Firo wallet on Tails OS can help onboard a broader audience, further strengthening the Firo community and its adoption in the cryptocurrency space.
